modules/cms-kit/src/Volo.CmsKit.Admin.Application/Volo/CmsKit/Admin/Blogs/BlogAdminAppService.cs

```diff
@@ -33,5 +40,17 @@ protected override async Task<IQueryable<Blog>> CreateFilteredQueryAsync(BlogGet
                         !input.Filter.IsNullOrWhiteSpace(),
                         x => x.Name.ToLower().Contains(input.Filter));
         }
+
+        public override async Task DeleteAsync(Guid id)
+        {
+            var hasPosts = await BlogPostRepository.ExistsAsync(id);
```

Comment:
I think this should be a domain rule instead of checking inside the application layer. What do you think @cotur?

 See https://docs.abp.io/en/abp/latest/Domain-Driven-Design-Implementation-Guide#business-logic-in-entities-requiring-external-services

Reply:
I agree with that, preventing deletion of blog is domain rule. It can't be changed according to applications. 
fyi @cotur